combustion-android-ble/src/main/java/inc/combustion/framework/ble/ProbeManager.kt

```diff
@@ -314,33 +388,42 @@ internal class ProbeManager(
             // remove this item from the list of firmware details for the network
             dfuDisconnectedNodeCallback(device.id)
 
-            // session info is now invalid
-            sessionInfo = null
-
-            // event out the invalid session info, fw version, and hardware revision.
-            _probe.value = _probe.value.copy(
-                baseDevice = _probe.value.baseDevice.copy(fwVersion = null, hwRevision = null, modelInformation = null),
-                sessionInfo = null,
-            )
+            // event out the invalid session info.
+            _probe.value = _probe.value.copy(sessionInfo = null)
         }
 
-        if(settings.meatNetEnabled) {
-            val meatNetState = if(arbitrator.isConnectedToMeatNet()) DeviceConnectionState.CONNECTED else DeviceConnectionState.DISCONNECTED
-
-            // if using meatnet, then we use the arbitrated connection state
-            _probe.value = _probe.value.copy(baseDevice = _probe.value.baseDevice.copy(connectionState = meatNetState))
-
+        // if don't have a route to send UART messages and we are currently in an active
+        // Upload State, coordinate with Log Manager that it can no longer expect log
+        // messages, and transition the upload state to Unavailable.  Null SessionInfo
+        // so that it can be requested again when a route is established
+        if(arbitrator.hasNoUartRoute && uploadState != ProbeUploadState.Unavailable) {
+            sessionInfo = null
+            uploadState = ProbeUploadState.Unavailable
```

Comment:
It looks like this will event out the probe, and you event again after this `if` block--just want to make sure that you don't expect any side effects of `logTransferCompleteCallback()` to happen before the probe state's evented out.

Reply:
let me check

Reply:
At this point no -- but it seems cleaner to make the callback before in case that assumption isn't true in the future.